text/0000-builder-spec.md

```diff
@@ -0,0 +1,295 @@
+# Meta
+[meta]: #meta
+- Name: Builder Spec
+- Start Date: 2020-09-11
+- Author(s): [@dfreilich](https://github.com/dfreilich)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+A Builder is a critical tool in the CNB ecosystem, and yet, is largely [unspecified][spec-issue-builder]. This RFC proposes that we should add a rigorous versioned specification of builders as an extension to the core specification, and clarify the existence of builders in the:
+* [distribution specification][distribution-spec]
+* [platform specification][platform-spec]
+
+# Definitions
+[definitions]: #definitions
+* **[Builder][builder-docs]** &rarr; A **builder** is an image that bundles all the bits and information on how to build your apps, such as buildpacks, an implementation of the lifecycle, and build-time environment that platforms may use when executing the lifecycle.
+* **[Buildpack][buildpacks-docs]** &rarr; A **buildpack** is a unit of work that inspects your app source code and formulates a plan to build and run your application.
+* **[Platform][platform-docs]** &rarr; A **platform** uses a lifecycle, buildpacks (packaged in a builder), and application source code to produce an OCI image.
+* **[Stack][stack-docs]** &rarr; A stack provides the buildpack lifecycle with build-time and run-time environments in the form of images.
+* **[Lifecycle][lifecycle-docs]** &rarr; The lifecycle orchestrates buildpack execution, then assembles the resulting artifacts into a final app image.
+* **Extension Spec** &rarr; An extension spec defines soemthing that is an optional add-on for Cloud Native Buildpacks systems, but not required to be considered `spec` compliant.
+
+# Motivation
+[motivation]: #motivation
+Builders are a vital part of the CNB ecosystem as it currently stands; they are used to distribute the lifecycle and buildpacks to every platform, including:
```

Comment:
```suggestion
Builders are a vital part of the CNB ecosystem as it currently stands; they are used to distribute the lifecycle and buildpacks to many platforms, including:
```

Reply:
Done in 211b74c7c7443b353cfa111cc15c92abb60369a9